backend/queues/printfulSyncProcessor.js

```diff
@@ -1,16 +1,43 @@
 const queues = require('./queues')
 
-const downloadProductData = require('../scripts/printful/downloadProductData')
-const writeProductData = require('../scripts/printful/writeProductData')
-const downloadPrintfulMockups = require('../scripts/printful/downloadPrintfulMockups')
-const resizePrintfulMockups = require('../scripts/printful/resizePrintfulMockups')
+const downloadProductData = require('../logic/printful/sync/downloadProductData')
+const writeProductData = require('../logic/printful/sync/writeProductData')
+const downloadPrintfulMockups = require('../logic/printful/sync/downloadPrintfulMockups')
+const resizePrintfulMockups = require('../logic/printful/sync/resizePrintfulMockups')
 
 const { Shop } = require('../models')
 
 const attachToQueue = () => {
   const queue = queues['printfulSyncQueue']
   queue.process(processor)
   queue.resume() // Start if paused
+
+  const updateSyncStatus = async (job, status, hasChanges) => {
+    if (!job.data) return
+
+    const { shopId } = job.data
+
+    if (!shopId) return
+
+    await Shop.update(
+      { printfulSyncing: status, hasChanges },
```

Comment:
hasChanges is passed as an arg to this functions. Yeah, I set it to true only when the job has completed.

Reply:
Sorry I'm still confused...  hasChanges is passed as false on line 29 below and not passed (undefined) on line 32, 36, 39

Reply:
Ah, good catch. It should `true` on line 29. Will change it.

As for the other three, Sequelize will ignore fields that are `undefined` by default. You'll have to use `null` if you want to unset a value.

Reply:
Overall I would think that the only time hasChange should be updated is when a sync is completed.
We never want to set it to false, since it could have been set to true due to a completely separate config change that requires the shop to get re-published. Only the publish method should ever set hasChanges to false.

Reply:
Sure, will change it.